src/client/debug/drive-robot.tsx

```diff
@@ -43,35 +57,73 @@ export function DriveRobot(props: DriveRobotProps) {
                     // tank-style drive
                     // deadzone and scaling to be more sensitive at lower values
                     const DEADZONE = 0.15;
-                    let leftPower = gamepad.axes[1] * -1;
-                    let rightPower = gamepad.axes[3] * -1;
-                    if (Math.abs(leftPower) < DEADZONE) leftPower = 0;
-                    if (Math.abs(rightPower) < DEADZONE) rightPower = 0;
-                    leftPower = Math.sign(leftPower) * Math.pow(leftPower, 2);
-                    rightPower =
-                        Math.sign(rightPower) * Math.pow(rightPower, 2);
-                    props.sendMessage(
-                        new DriveRobotMessage(
-                            props.robotId,
-                            leftPower,
-                            rightPower,
-                        ),
-                    );
+                    let padLeftPower = gamepad.axes[1] * -1;
+                    let padRightPower = gamepad.axes[3] * -1;
+                    if (Math.abs(padLeftPower) < DEADZONE) padLeftPower = 0;
+                    if (Math.abs(padRightPower) < DEADZONE) padRightPower = 0;
+                    padLeftPower =
+                        Math.sign(padLeftPower) * Math.pow(padLeftPower, 2);
+                    padRightPower =
+                        Math.sign(padRightPower) * Math.pow(padRightPower, 2);
+                    if (
+                        prevPadLeft === 0 &&
+                        prevPadRight === 0 &&
+                        padLeftPower === 0 &&
+                        padRightPower === 0
+                    ) {
+                        continue;
+                    }
```

Comment:
`===` comparisons with floats is generally dubious, but I guess it's okay here? It depends on if we get joystick jitter or not.

Reply:
Pretty sure the jitter is handled with the joystick deadzone logic. Colin and I did pretty thorough testing of output when driving with joysticks and all seemed ok.

Reply:
this is true. Colin and I played with it, whenever you hold your thumb in a single location, it seemed _mostly_ stable. May be worth adding more jitter logic in the future

Reply:
it fixed now